torch/_refs/__init__.py

```diff
@@ -510,6 +512,38 @@ def _isnan(a: TensorLikeType) -> TensorLikeType:
 )
 
 
+@register_decomposition(torch.ops.aten.nan_to_num)
+@out_wrapper
+@elementwise_type_promotion_wrapper(
```

Comment:
Does nan_to_num meaningfully participate in type promotion?

Reply:
I see this failure `Dtypes torch.float32 and torch.bool are not equal!` on test_python_ref__refs_nan_to_num_cpu_bool without type promotion.